subsys/pm/state.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <pm/state.h>
+#include <toolchain.h>
+
+BUILD_ASSERT(DT_NODE_EXISTS(DT_PATH(cpus)),
+	     "cpus node not defined in Devicetree");
+
+/**
+ * Check CPU power state consistency.
+ *
+ * @param i Power state index.
+ * @param node_id CPU node identifier.
+ */
+#define CHECK_POWER_STATE_CONSISTENCY(i, node_id)			       \
+	BUILD_ASSERT(							       \
+		DT_PROP_BY_PHANDLE_IDX_OR(node_id, cpu_power_states, i,	       \
+					  min_residency_us, 0U) >=	       \
+		DT_PROP_BY_PHANDLE_IDX_OR(node_id, cpu_power_states, i,	       \
```

Comment:
I believe the assert here should be `min_residency_us > exit_latency_us`. Does it make any sense having `min_residency_us == exit_latency_us` ?

Reply:
This is keeping current behavior, which btw you suggested here: https://github.com/zephyrproject-rtos/zephyr/pull/41124#discussion_r768333685

Reply:
yep yep yep. Missed the assert, it continues to make sense the `>=` :)